src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs

```diff
@@ -195,15 +195,23 @@ private static void WriteConstructor(CodeWriter writer, ObjectType schema)
 
                 using (writer.Scope())
                 {
-                    writer.WriteParameterNullChecks(constructor.Parameters);
+                    foreach (var parameter in constructor.Parameters)
+                    {
+                        if (parameter.DefaultValue == null)
+                        {
+                            writer.WriteParameterNullChecks(parameter);
+                        }
+                    }
+                    writer.Line();
 
                     foreach (var initializer in constructor.Initializers)
                     {
                         writer.Append($"{initializer.Property.Declaration.Name} = ")
-                            .WriteConversion(initializer.Value.Type, initializer.Property.Declaration.Type, w=>w.WriteReferenceOrConstant(initializer.Value));
+                            .WriteConversion(initializer.Value.Type, initializer.Property.Declaration.Type, w => w.WriteReferenceOrConstant(initializer.Value));
 
                         if (initializer.DefaultValue != null &&
-                            !initializer.Value.Type.IsValueType)
+                        (!initializer.Value.Type.IsValueType ||
```

Comment:
This condition looks suspiciously complicated. Do we need to change some logic around discriminator initializer in ObjectType.cs instead?

Reply:
Fixed!